src/feature-test/java/uk/gov/ida/verifyserviceprovider/configuration/MsaMetadataFeatureTest.java

```diff
@@ -36,48 +36,64 @@
         wireMockServer.start();
         hubServer.serveDefaultMetadata();
 
+        KeyStoreResource verifyMetadataKeystoreResource = aKeyStoreResource()
+                .withCertificate("VERIFY-FEDERATION", aCertificate().withCertificate(METADATA_SIGNING_A_PUBLIC_CERT).build().getCertificate())
+                .build();
+        verifyMetadataKeystoreResource.create();
         KeyStoreResource verifyHubKeystoreResource = aKeyStoreResource()
-            .withCertificate("VERIFY-FEDERATION", aCertificate().withCertificate(METADATA_SIGNING_A_PUBLIC_CERT).build().getCertificate())
-            .build();
+                .withCertificate("VERIFY-FEDERATION", aCertificate().withCertificate(CACertificates.TEST_CORE_CA).build().getCertificate())
+                .build();
         verifyHubKeystoreResource.create();
         this.applicationTestSupport = new DropwizardTestSupport<>(
             VerifyServiceProviderApplication.class,
             "verify-service-provider.yml",
             config("server.connector.port", "0"),
-            config("verifyHubConfiguration.metadata.uri", format("http://localhost:%s/SAML2/metadata", hubServer.port())),
+            config("verifyHubConfiguration.environment", "CUSTOM"),
             config("msaMetadata.uri", getMsaMetadataUrl()),
-            config("verifyHubConfiguration.metadata.expectedEntityId", HUB_ENTITY_ID),
             config("msaMetadata.expectedEntityId", MockMsaServer.MSA_ENTITY_ID),
-            config("verifyHubConfiguration.metadata.trustStore.path", verifyHubKeystoreResource.getAbsolutePath()),
-            config("verifyHubConfiguration.metadata.trustStore.password", verifyHubKeystoreResource.getPassword())
+            config("serviceEntityIds", "[\"http://some-service-entity-id\"]"),
+            config("samlSigningKey", TEST_RP_PRIVATE_SIGNING_KEY),
+            config("samlPrimaryEncryptionKey", TEST_RP_PRIVATE_ENCRYPTION_KEY)
         );
 
-        environmentHelper.setEnv(new HashMap<String, String>() {{
-            put("VERIFY_ENVIRONMENT", "COMPLIANCE_TOOL");
-            put("MSA_METADATA_URL", "some-msa-metadata-url");
-            put("MSA_ENTITY_ID", "some-msa-entity-id");
-            put("SERVICE_ENTITY_IDS", "[\"http://some-service-entity-id\"]");
-            put("SAML_SIGNING_KEY", TEST_RP_PRIVATE_SIGNING_KEY);
-            put("SAML_PRIMARY_ENCRYPTION_KEY", TEST_RP_PRIVATE_ENCRYPTION_KEY);
-        }});
+        new HashMap<String, String>() {{
+            put(HUB_SSO_URL, String.format("http://localhost:%s/SAML2/SSO", hubServer.port()));
+            put(HUB_METADATA_URL, String.format("http://localhost:%s/SAML2/metadata", hubServer.port()));
+            put(HUB_EXPECTED_ENTITY_ID, HUB_ENTITY_ID);
+            put(METADATA_TRUSTSTORE_PATH, verifyMetadataKeystoreResource.getAbsolutePath());
+            put(HUB_TRUSTSTORE_PATH, verifyHubKeystoreResource.getAbsolutePath());
+            put(IDP_TRUSTSTORE_PATH, verifyHubKeystoreResource.getAbsolutePath());
```

Comment:
Do we need a separate truststore containing a root CA and a test idp ca certificates for idp?